youtube_dl/extractor/npo.py

```diff
@@ -184,21 +186,29 @@ def _real_extract(self, url):
         return self._get_info(url, video_id) or self._get_old_info(video_id)
 
     def _get_info(self, url, video_id):
-        token = self._download_json(
+        _, xsrf_token_response = self._download_webpage_handle(
             'https://www.npostart.nl/api/token', video_id,
             'Downloading token', headers={
                 'Referer': url,
                 'X-Requested-With': 'XMLHttpRequest',
-            })['token']
+            })
+        cookies = compat_cookies_SimpleCookie()
+        cookies.load(xsrf_token_response.headers['Set-Cookie'])
+        cookies = dict((k, v.value) for k, v in cookies.items())
+        xsrf_token = cookies['XSRF-TOKEN']
 
         player = self._download_json(
             'https://www.npostart.nl/player/%s' % video_id, video_id,
-            'Downloading player JSON', data=urlencode_postdata({
+            'Downloading player JSON',
+            headers={
+                'x-xsrf-token': compat_urllib_parse_unquote_plus(xsrf_token)
```

Comment:
Use the built-in cookie jar:
```suggestion
        xsrf_token = next((c.value for c in self.cookiejar if c.name == 'XSRF-TOKEN'), '')
        if xsrf_token:
            xsrf_token = compat_urllib_parse_unquote_plus(xsrf_token)
        else:
            self.report_warning('Failed to get XSRF token', video_id=video_id)

        player = self._download_json(
            'https://www.npostart.nl/player/%s' % video_id, video_id,
            'Downloading player JSON',
            headers={
                'x-xsrf-token': xsrf_token,
```
I'm not sure about the unquoting here but don't have a working URL to test it.